### DISCUSSION TITLE
Command line tool update : cmake don't like

### DISCUSSION THREAD

**Discussion by vreville** [idefix-code/idefix#271] [General]
Hi, 

I updated the command line tools on my mac os Sonoma 14.7, and apparently cmake is not liking it. I have the following message : 

> ```
> CMake Error at src/kokkos/cmake/kokkos_test_cxx_std.cmake:91 (MESSAGE):
>   C++17-compliant compiler detected, but unable to compile C++17 or later
>   program.  Verify that AppleClang:16.0.0.16000026 is set up correctly (e.g.,
>   check that correct library headers are being used).
> 
>   Failing output:
> 
>    Change Dir: '/Users/vreville/Codes/idefix_setups_gitlab/TubeSection/build/kokkos/corner_cases/CMakeFiles/CMakeTmp'
> 
>   
> 
>   Run Build Command(s): /usr/local/Cellar/cmake/3.30.3/bin/cmake -E env
>   VERBOSE=1 /usr/bin/make -f Makefile cmTC_09cb0/fast
> 
>   /Library/Developer/CommandLineTools/usr/bin/make -f
>   CMakeFiles/cmTC_09cb0.dir/build.make CMakeFiles/cmTC_09cb0.dir/build
> 
>   Building CXX object CMakeFiles/cmTC_09cb0.dir/cplusplus17.cpp.o
> 
>   /Library/Developer/CommandLineTools/usr/bin/c++ -std=gnu++17 -isysroot
>   /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk
>   -mmacosx-version-min=14.7 -MD -MT
>   CMakeFiles/cmTC_09cb0.dir/cplusplus17.cpp.o -MF
>   CMakeFiles/cmTC_09cb0.dir/cplusplus17.cpp.o.d -o
>   CMakeFiles/cmTC_09cb0.dir/cplusplus17.cpp.o -c
>   /Users/vreville/Codes/idefix/src/kokkos/cmake/compile_tests/cplusplus17.cpp
> 
> 
> 
>   /Users/vreville/Codes/idefix/src/kokkos/cmake/compile_tests/cplusplus17.cpp:17:10:
>   fatal error: 'type_traits' file not found
> 
>      17 | #include <type_traits>
>         |          ^~~~~~~~~~~~~
> 
>   1 error generated.
> 
>   make[1]: *** [CMakeFiles/cmTC_09cb0.dir/cplusplus17.cpp.o] Error 1
> 
>   make: *** [cmTC_09cb0/fast] Error 2
> 
> 
> 
> Call Stack (most recent call first):
>   src/kokkos/cmake/kokkos_tribits.cmake:202 (INCLUDE)
>   src/kokkos/CMakeLists.txt:226 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
> ```

I know it's not really relaled to Idefix, but I am not sure how to deal with this. I have a cmake from homebrew and I never had to build it before. 

Any help appreciated!

Victor

**Comment by glesur** [comment]
Did you start from a clean environement after this update (i.e., make clean, rm CMakeCache.txt, new terminal, possibly even a reboot?)

**Comment by vreville** [comment]
Yes, I tried to remove all cmake files and folders in the compilation repo, and just rebooted the computer and I still have this. It may be a bug from the Apple update.

**Reply by glesur** [reply]
Can you check that `/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk` is a directory that exists and contains your commandline tools? If not, try to set [CMAKE_OSX_SYSROOT](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html#variable:CMAKE_OSX_SYSROOT) to the right path before invoking cmake?

**Comment by vreville** [comment]
I tried to 

> export CMAKE_OSX_SYSROOT = /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk

also 

> export CMAKE_OSX_SYSROOT = /Library/Developer/CommandLineTools/SDKs/MacOSX14.5.sdk

and it did not work. Eventually, I just rm CommandLineTools and reinstalled it, with the latest version being 14.4. It works now.

**Reply by glesur** [reply]
I assume we can now close the discussion?

**Comment by vreville** [comment]
Yes, I guess the advice is do not update CLT now.

Le jeu. 3 oct. 2024 à 08:31, Geoffroy Lesur ***@***.***> a
écrit :
